tools/proxy-endpoint-unifier/utils.py

```diff
@@ -0,0 +1,506 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import configparser
+import os
+import sys
+import xmltodict
+import json
+import shutil
+import zipfile
+
+
+def parse_config(config_file):
+    config = configparser.ConfigParser()
+    config.read(config_file)
+    return config
+
+
+def get_proxy_endpoint_count(cfg):
+    try:
+        proxy_endpoint_count = cfg.getint('common', 'proxy_endpoint_count')
+        if not (proxy_endpoint_count > 0 and proxy_endpoint_count <= 5):
+            print('ERROR: Proxy Endpoints should be > Zero(0)  &  < Five(5)')
+            sys.exit(1)
+    except ValueError:
+        print('proxy_endpoint_count should be Numberic')
+        sys.exit(1)
+    return proxy_endpoint_count
+
+
+def create_dir(dir):
+    try:
+        os.makedirs(dir)
+    except FileExistsError:
+        print(f"INFO: {dir} already exists")
+
+
+def list_dir(dir, isok=False):
+    try:
+        return os.listdir(dir)
+    except FileNotFoundError:
+        if isok:
+            print(f"Ignoring : Directory \"{dir}\" not found")
+            return []
+        print(f"ERROR: Directory \"{dir}\" not found")
+        sys.exit(1)
+
+
+def get_proxy_entrypoint(dir):
+    files = list_dir(dir)
+    ent = []
+    for eachfile in files:
+        if eachfile.endswith(".xml"):
+            ent.append(eachfile)
+    if len(ent) == 1:
+        return os.path.join(dir, ent[0])
+    else:
+        if len(ent) > 1:
+            print(f"ERROR: Directory \"{dir}\" contains multiple xml files at root")  # noqa
+        else:
+            print(f"ERROR: Directory \"{dir}\" has no xml file at root")  # noqa
+    return None
+
+
+def parse_json(file):
+    try:
+        with open(file) as fl:
+            doc = json.loads(fl.read())
+        return doc
+    except FileNotFoundError:
+        print(f"ERROR: File \"{file}\" not found")
+    return {}
+
+
+def parse_xml(file):
+    try:
+        with open(file) as fl:
+            doc = xmltodict.parse(fl.read())
+        return doc
+    except FileNotFoundError:
+        print(f"ERROR: File \"{file}\" not found")
+    return {}
+
+
+def write_json(file, data):
+    try:
+        with open(file, 'w') as fl:
+            fl.write(json.dumps(data, indent=2))
+    except FileNotFoundError:
+        print(f"ERROR: File \"{file}\" not found")
+        return False
+    return True
+
+
+def write_xml_from_dict(file, data):
+    try:
+        with open(file, 'w') as fl:
+            fl.write(xmltodict.unparse(data, pretty=True))
+    except FileNotFoundError:
+        print(f"ERROR: File \"{file}\" not found")
+        return False
+    return True
+
+
+def parse_proxy_root(dir):
+    file = get_proxy_entrypoint(dir)
+    if file is None:
+        return {}
+    doc = parse_xml(file)
+    return doc
+
+
+def read_proxy_artifacts(dir, entrypoint):
```

Comment:
What about resources?

Reply:
For this MVP we are only dealing with Policies and proxy endpoints